blueocean-github-pipeline/src/main/java/io/jenkins/blueocean/blueocean_github_pipeline/HttpRequest.java

```diff
@@ -0,0 +1,141 @@
+package io.jenkins.blueocean.blueocean_github_pipeline;
+
+import com.fasterxml.jackson.databind.JsonMappingException;
+import io.jenkins.blueocean.commons.ServiceException;
+import org.apache.commons.io.IOUtils;
+import org.kohsuke.github.HttpConnector;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.lang.reflect.Array;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Collections;
+import java.util.zip.GZIPInputStream;
+
+/**
+ * @author Vivek Pandey
+ */
+class HttpRequest {
+    private final String method;
+    private final String url;
+    private String contentType = "application/json; charset=utf-8";
+    private Object body;
+
+    private String authorization;
+
+    private HttpRequest(String method, String url) {
+        this.method = method;
+        this.url = url;
+    }
+
+    public static HttpRequest get(String url){
+        return new HttpRequest("GET", url);
+    }
+
+    public static HttpRequest head(String url){
+        return new HttpRequest("HEAD", url);
+    }
+
+    public static HttpRequest post(String url){
+        return new HttpRequest("POST", url);
+    }
+
+    public static HttpRequest put(String url){
+        return new HttpRequest("PUT", url);
+    }
+
+    public HttpRequest with(String contentType){
+        this.contentType = contentType;
+        return this;
+    }
+
+    public HttpRequest withAuthorization(String authorization){
+        this.authorization = authorization;
+        return this;
+    }
+
+    public HttpRequest withBody(Object body){
+        this.body = body;
+        return this;
+    }
+
+    public <T> T to(Class<T> type) throws IOException {
+        HttpURLConnection connection = connect();
+        if (methodNeedsBody()) {
+            connection.setDoOutput(true);
+            connection.setRequestProperty("Content-type", contentType);
+            if (body == null) {
+                GithubScm.om.writeValue(connection.getOutputStream(), Collections.emptyMap());
```

Comment:
Ha ha yeah references this ObjectMapper from that class:) Good catch, that ObjectMapper should have belonged to HttpRequest. There is another PR I am working on where I am touching that class as that needs cleanup anyways.